File: csharp/Behavioral/20-Strategy/Model/JobCard.cs
namespace Strategy;

public class JobCard
{
    public string JobId { get; } = Guid.NewGuid().ToString("N")[..8];

    public string Title { get; init; } = string.Empty;
    public string Category { get; init; } = string.Empty;
    public decimal QuotedPrice { get; init; }
    public decimal HoursWorked { get; set; }
    public bool IsAfterHours { get; set; }

    public override string ToString() =>
        $"Job #{JobId} '{Title}' [{Category}] — {HoursWorked}h";
}

File: typescript/Behavioral/20-Strategy/Model/JobCard.ts
export class JobCard {
  readonly jobId: string = Math.random().toString(36).slice(2, 10);

  hoursWorked: number = 0;
  isAfterHours: boolean = false;

  constructor(
    readonly title: string,
    readonly category: string,
    readonly quotedPrice: number = 0,
  ) {}

  toString(): string {
    return `Job #${this.jobId} '${this.title}' [${this.category}] — ${this.hoursWorked}h`;
  }
}
